packages/frontend/src/components/Buttons/BackButton.tsx

```diff
@@ -1,15 +1,29 @@
+import { useCallback } from 'react'
+import { useNavigate } from 'react-router-dom'
 import { ArrowLeftIcon } from 'src/components/Icons/ArrowLeftIcon'
 import { Colors } from 'src/styles/colors'
 import styled from 'styled-components'
 
 interface BackButtonProps {
-  view: number
-  setView: (view: number) => void
+  view?: number
+  setView?: (view: number) => void
+  url?: string
 }
 
-export function BackButton({ view, setView }: BackButtonProps) {
+export function BackButton({ view, setView, url }: BackButtonProps) {
+  const navigate = useNavigate()
+
+  const goBack = useCallback(() => {
+    if (view && setView) {
+      setView(view - 1)
+    }
+    if (url) {
+      navigate(`${url}`)
```

Comment:
Shouldn't it just say-?
```suggestion
      navigate(url)
```